docs/FRONTEND.md

```diff
@@ -0,0 +1,203 @@
+# CSS at Guardian Membership
+
+---
+
+**Table of contents**
+
+* [Architecture](#architecture)
+* [Naming Conventions](#naming)
+* [BEM Recommendations](#bem)
+* [JavaScript](#javascript)
+* [Vendor Prefixes](#prefixes)
+* [Further Reading](#reading)
+
+<a name="architecture"></a>
+## Architecture
+
+We use Sass for our stylesheets and break styles down into the following groups:
+
+1. **Settings**: Global variables (colours, breakpoints etc.)
+2. **Mixins**: Functions and mixins
+3. **Base**: Un-classed HTML elements and resets (normalize)
+4. **Helpers**: Common design patterns and traits (grid, text, heading traits)
+5. **Components**: Styled objects, chunks of UI
+5. **Overrides (trumps)**: Helpers and overrides, (utilities class, responsive visibility helpers, colours).
+
+<a name="naming"></a>
+## Naming conventions:
+
+We largely use the following naming conventions in our CSS:
+
+- We (sparingly) use low-level helper classes for commmon traits, e.g., headings, utilities
+- We use a version of BEM to distinguish between components
+- We use state classes (e.g., `is-toggled`) to style components affecteder by JavaScript
+
```

Comment:
I remember this from our conversation, maybe these are just standard CS rules:
- If it is a component in markup then the Bem classes should display this to the reader
- Componentize markup where possible for reuse, until it feels "right" not to big not to small